stanford-sw/src/edu/stanford/Item.java

```diff
@@ -14,545 +14,548 @@
+  private final String library;
+  private final String itemType;
+  private final String publicNote;
+  private final boolean shouldBeSkipped;
+  private final boolean hasGovDocLoc;
+  private final boolean isOnline;
+  private final boolean hasShelbyLoc;
+  private final boolean hasBizShelbyLoc;
+  private final boolean hasLaneLoc;
+  private final boolean hasArtLockedLoc;
+
+  /* normal instance variables */
+  private CallNumberType callnumType;
+  private String homeLoc;
+  private String currLoc;
+  private String normCallnum;
+  private boolean isOnOrder = false;
+  private boolean isInProcess = false;
+  private boolean hasIgnoredCallnum = false;
+  private boolean hasBadLcLaneCallnum = false;
+  private boolean isMissingLost = false;
+  private boolean hasSeparateBrowseCallnum = false;
+  /** call number with volume suffix lopped off the end.  Used to remove
+   * noise in search results and in browsing */
+  private String loppedCallnum = null;
+  /** set when there is a callnumber for browsing different from that in the 999 */
+  private String browseCallnum = null;
+
+  /** sortable version of lopped call number */
+  private String loppedShelfkey = null;
+
+  /** reverse sorted version of loppedShelfkey - the last call number shall
+   * be first, etc. */
+  private String reverseLoppedShelfkey = null;
+
+  /** sortable full call number, where, for serials, any volume suffix will sort
+   * in descending order.  Non-serial volumes will sort in ascending order. */
+  private String callnumVolSort = null;
+
+
+
+  /**
+   * initialize object from 999 DataField, which has the following subfields
+   * <ul>
+   *   <li>a - call num</li>
+   *   <li>i - barcode</li>
+   *   <li>k - current location</li>
+   *   <li>l - home location</li>
+   *   <li>m - library code</li>
+   *   <li>o - public note</li>
+   *   <li>t - item type</li>
+   *   <li>w - call num type</li>
+   *  </ul>
+   */
+  public Item(DataField f999, String recId) {
+    // set all the immutable variables
+    this.recId = recId;
+    barcode = MarcUtils.getSubfieldTrimmed(f999, 'i');
+    currLoc = MarcUtils.getSubfieldTrimmed(f999, 'k');
+    homeLoc = MarcUtils.getSubfieldTrimmed(f999, 'l');
+    library = MarcUtils.getSubfieldTrimmed(f999, 'm');
+    publicNote = MarcUtils.getSubfieldTrimmed(f999, 'o');
+    itemType = MarcUtils.getSubfieldTrimmed(f999, 't');
+    String scheme = MarcUtils.getSubfieldTrimmed(f999, 'w');
+    String rawCallnum = MarcUtils.getSubfieldTrimmed(f999, 'a');
+
+    if (StanfordIndexer.SKIPPED_LOCS.contains(currLoc)
+          || StanfordIndexer.SKIPPED_LOCS.contains(homeLoc)
+          || itemType.equals("EDI-REMOVE")
+          // SW-849 skip PHYSICS items that aren't location PHYSTEMP
+          || (library.equals("PHYSICS") && (!homeLoc.equals("PHYSTEMP") && !currLoc.equals("PHYSTEMP"))))
+      shouldBeSkipped = true;
+    else if (StanfordIndexer.CLOSED_LIBS.contains(library) && (StanfordIndexer.RESV_LOCS.contains(currLoc)))
```

Comment:
@ndushay it's specifically these next few lines that I changed (in addition to turning tabs into spaces!).